byway/src/main/java/com/google/sps/servlets/DestinationsServlet.java

```diff
@@ -24,62 +26,53 @@
 
   private final Gson gson = new Gson();
   private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-  private Key userKey;
-
-  @Override
-  public void init() {
-    Entity userEntity = new Entity("UserInputs");
-    userEntity.setProperty("start", "");
-    userEntity.setProperty("destinations", new ArrayList<String>());
-    datastore.put(userEntity);
-    userKey = userEntity.getKey();
-  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Key tripKey = KeyFactory.stringToKey(request.getParameter("tripKey"));
     Entity entity;
     try {
-      entity = datastore.get(userKey);
+      entity = datastore.get(tripKey);
     } catch (EntityNotFoundException e) {
-      UserLocations userLocations = new UserLocations("", new ArrayList<String>());
-      response.setContentType("application/json;");
-      response.getWriter().println(gson.toJson(userLocations));
-      logger.atInfo().withCause(e).log("Unable to find UserLocations Entity %s", userKey);
+      logger.atInfo().withCause(e).log("Unable to find Trip Entity %s", tripKey);
+      response.setStatus(HttpServletResponse.SC_NOT_FOUND);
       return;
     }
     String start = (String) entity.getProperty("start");
```

Comment:
Consider making a static method of `Trip` to retrieve a trip (or null) based on a `Key` or `String` representation of a `Key`. That way logic like this becomes:

```java
Trip trip = Trip.getTrip(datastore, request.getParameter("tripKey"));
if (trip == null) {
  response.setStatus(HttpServletResponse.SC_NOT_FOUND);
  return;
}
// ... rest of operations on the trip.
```

This becomes a huge benefit if you need to `getTrip` in many places! Only one copy of that code, and you don't have to worry about `try/catch` everywhere, just a simple `null` check.

Reply:
done